src/validation.h

```diff
@@ -455,6 +458,16 @@ class BlockManager
     //! Returns last CBlockIndex* that is a checkpoint
     CBlockIndex* GetLastCheckpoint(const CCheckpointData& data) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
+    /**
+     * Map from external index name to most recent block the index can tolerate being pruned.
+     *
+     * @note Internally, only blocks at height (block->nHeight - PRUNE_BLOCKER_BUFFER) and
+     * below will be pruned, but callers should avoid assuming any particular buffer size.
+     */
+    std::unordered_map<std::string, const CBlockIndex*> m_prune_blockers GUARDED_BY(::cs_main);
```

Comment:
Better to just use a height here, I think. See 598b0d8dbf5793c60c63dc2a181a49291fe8011b

Reply:
Added this